src/main/java/hudson/plugins/accurev/AccurevLauncher.java

```diff
@@ -432,13 +432,14 @@ public static String getAccurevExe(
   }
 
   private static Integer runCommandToCompletion( //
-      final ProcStarter starter, //
-      final Lock synchronizationLockObjectOrNull)
+      @Nonnull final ProcStarter starter, //
+      final ReentrantLock synchronizationLockObjectOrNull)
       throws IOException, InterruptedException {
+    if (synchronizationLockObjectOrNull != null) {
+      synchronizationLockObjectOrNull.lockInterruptibly();
```

Comment:
Is it wise to move it outside the try block?

Reply:
lockInterruptibly() outside of the try block is correct.  The lock is either successful or not, and we don't want it to try to unlock in the finally block if something goes wrong with getting the lock in the first place.